src/debtManagers/HookAMM.sol

```diff
@@ -0,0 +1,238 @@
+        dola.transferFrom(msg.sender, address(this), exactDolaIn);
+        uint dbrBal = dbr.balanceOf(address(this));
+        if(exactDbrOut > dbrBal)
+            dbr.mint(address(this), exactDbrOut - dbrBal);
+        uint exactDbrOutAfterFee = exactDbrOut * (1e18 - dbrBuyFee) / 1e18;
+        feesAccrued += exactDbrOut - exactDbrOutAfterFee;
+        dbr.transfer(to, exactDbrOutAfterFee);
+        emit BuyDBR(msg.sender, to, exactDolaIn, exactDbrOut);
+    }
+
+    function burnDBR(uint exactDolaIn, uint exactDbrBurn) external buyHook {
+        _invariantCheck(exactDolaIn, exactDbrBurn, getDbrReserve());
+        dola.transferFrom(msg.sender, address(this), exactDolaIn);
+        dbr.burn(exactDbrBurn);
+        emit Burn(msg.sender, exactDolaIn, exactDbrBurn);
+    }
+
+    function buyDola(uint exactDbrIn, uint exactDolaOut, address to) external buyHook {
+        require(to != address(0), "Zero address");
+        uint k = getK();
+        uint dolaReserve = getDolaReserve();
+        uint dbrReserve = k / dolaReserve;
+        _invariantCheck(dbrReserve - exactDbrIn, dolaReserve - exactDolaOut, getDolaReserve());
+        dbr.transferFrom(msg.sender, address(this), exactDolaOut);
+        dola.transfer(to, exactDbrIn);
+        emit BuyDOLA(msg.sender, to, exactDbrIn, exactDolaOut);
+    }
+
+    function _invariantCheck(uint newDolaReserve, uint newDbrReserve, uint k) internal view {
+        if(newDolaReserve * newDbrReserve < k){
+            if(1e18 * newDbrReserve / newDolaReserve > maxDolaPrice){
+                uint maxDolaReserve = sqrt(1e18 * k / maxDolaPrice);
+                uint maxDbr = k / maxDolaReserve;
+                uint excessDola = newDolaReserve - maxDolaReserve;
+                uint excessDbr = newDbrReserve - maxDbr;
+                if(1e18 * excessDbr / excessDola >= maxDolaPrice) revert Invariant();
+            } else {
+                revert Invariant();
+            }
+        }
+    }
+
+    /**
+     * @dev Sets a new pending governance address.
+     * @param _gov The address of the new pending governance.
+     */
+    function setPendingGov(address _gov) external onlyGov {
+        pendingGov = _gov;
+    }
+
+    /**
+     * @dev Allows the pending governance to accept its role.
+     */
+    function acceptGov() external {
+        require(msg.sender == pendingGov, "ONLY PENDINGGOV");
+        gov = pendingGov;
+        pendingGov = address(0);
+    }
+
+    /**
+     * @dev Allows governance to sweep any ERC20 token from the contract.
+     * @dev Excludes the ability to sweep DBR tokens.
+     * @param token The address of the ERC20 token to sweep.
+     * @param amount The amount of tokens to sweep.
+     * @param to The recipient address of the swept tokens.
+     */
+    function sweep(address token, uint amount, address to) public onlyGov {
+        require(address(dbr) != token, "Not authorized");
+        IERC20(token).transfer(to, amount);
+    }
+
+    function harvest() public {
```

Comment:
Consider removing the feeRecipient role and removing this function. We can simply burn the excess DBR fees